src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -15,8 +15,9 @@ namespace System.Formats.Tar
     public abstract partial class TarEntry
     {
         internal TarHeader _header;
+
         // Used to access the data section of this entry in an unseekable file
-        private TarReader? _readerOfOrigin;
+        internal TarReader? _readerOfOrigin;
```

Comment:
```suggestion
        private protected TarReader? _readerOfOrigin;
```

Reply:
I need it to be `internal` because if I use the suggested accessibility modifiers, then I cannot pass the reader of origin from `other` anymore. I get the following errors:

```
D:\runtime\src\libraries\System.Formats.Tar\src\System\Formats\Tar\V7TarEntry.cs(34):
CS1540: Cannot access protected member 'TarEntry._readerOfOrigin' via a qualifier of type 'TarEntry'; the qualifier must be of type 'V7TarEntry' (or derived from it)

D:\runtime\src\libraries\System.Formats.Tar\src\System\Formats\Tar\UstarTarEntry.cs(39):
CS1540: Cannot access protected member 'TarEntry._readerOfOrigin' via a qualifier of type 'TarEntry'; the qualifier must be of type 'UstarTarEntry' (or derived from it)

D:\runtime\src\libraries\System.Formats.Tar\src\System\Formats\Tar\PaxTarEntry.cs(97):
CS1540: Cannot access protected member 'TarEntry._readerOfOrigin' via a qualifier of type 'TarEntry'; the qualifier must be of type 'PaxTarEntry' (or derived from it)

D:\runtime\src\libraries\System.Formats.Tar\src\System\Formats\Tar\PaxTarEntry.cs(40):
CS1540: Cannot access protected member 'TarEntry._readerOfOrigin' via a qualifier of type 'TarEntry'; the qualifier must be of type 'GnuTarEntry' (or derived from it)
```

Reply:
Ah, got it. It is because you are accessing some other TarEntry's field.